### PR TITLE
Fallback Authentication Provider

### PR DESCRIPTION
**Pull request by Pfuenzle**
With this commit, a Fallback Provider can be set in the config via `<DefaultProvider>NAMEOFPROVIDER</DefaultProvider>`.

The set provider then gets assigned to the user after he has been logged in. If it is not set, nothing is changed.

With this, an user can login via SSO but is still able to log in via other providers on a later moment via username/password. This could also be changed to always set the Default Provider to the normal on (using the Jellyfin DB) if needed.

### CODE REVIEW

File: SSO-Auth/Api/SSOController.cs [hunk 1]
```diff
@@ -246,6 +246,15 @@ public async Task<ActionResult> OIDAuth([FromBody] AuthResponse response)
                     {
                         var authenticationResult = await Authenticate(kvp.Value.Username, kvp.Value.Admin, config.EnableAuthorization, config.EnableAllFolders, kvp.Value.Folders.ToArray(), response)
                             .ConfigureAwait(false);
+                            
+                        if(config.DefaultProvider != "")
```
Comment (9p4) [inline SSO-Auth/Api/SSOController.cs:250]: This should not be here, and instead should be moved into the `Authenticate` function.

File: SSO-Auth/Api/SSOController.cs [hunk 2]
```diff
@@ -384,6 +393,15 @@ public async Task<ActionResult> SamlAuth([FromBody] AuthResponse response)
                 }
                 var authenticationResult = await Authenticate(samlResponse.GetNameID(), isAdmin, config.EnableAuthorization, config.EnableAllFolders, folders.ToArray(), response)
                     .ConfigureAwait(false);
+                    
+                if(config.DefaultProvider != "")
```
Comment (9p4) [inline SSO-Auth/Api/SSOController.cs:397]: This should not be here, and instead should be moved into the `Authenticate` function.

**Review by 9p4 — CHANGES_REQUESTED** [review]

**Comment by Pfuenzle** [comment]
I also thought that, but in Authenticate() the "AuthenticateDirect" function gets called on return and I guessed that changing the Auth Provider to something else before that call would interfere with the login process.
But if it doesnt, Ill try to change it

**Comment by 9p4** [comment]
I would say to add another parameter to AuthenticateDirect and have the code in one place to prevent duplication.

**Comment by 9p4** [comment]
@Pfuenzle any update?

**Comment by Pfuenzle** [comment]
Works fine now from my testing, sorry it took so long. also all warnings and errors should be gone

**Review by 9p4 — APPROVED** [review]

**Comment by 9p4** [comment]
I'm not going to merge the build script because it seems to be specific to your machine.

**Comment by Pfuenzle** [comment]
Oh yes, I did not mean to include it, that was just for me to build it, missed the file when committing.
Thanks for merging it 👍

**Comment by 9p4** [comment]
Thanks for your contribution! Once I get this automated test system running, I'll push out a release.

**Pull request merged**